iac-extensions/terraform/src/test/resources/checks/Azure/AnonymousAccessToResourceCheck/data_factory_linked_service.tf

```diff
@@ -0,0 +1,36 @@
+# Noncompliant@+1 {{Omitting the 'basic_authentication' block disables authentication. Make sure it is safe here.}}
+resource "azurerm_data_factory_linked_service_odata" "sensitive" {
+#        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+}
+
+resource "azurerm_data_factory_linked_service_odata" "compliant" { # Compliant
+  basic_auth {
+    username = local.creds.username
+    password = local.creds.password
+  }
+}
+
+resource "azurerm_data_factory_linked_service_odata" "compliant_empty" { # Compliant
+  basic_auth {
+  }
+}
+
+resource "azurerm_data_factory_linked_service_sftp" "example2" {
+  # Noncompliant@+1 {{Make sure that authorizing anonymous access is safe here.}}
+  authentication_type = "Anonymous"
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+}
+
+resource "azurerm_data_factory_linked_service_sftp" "compliant" { # Compliant
+  authentication_type = "Basic"
+  username            = local.creds.username
+  password            = local.creds.password
+}
+
+resource "azurerm_data_factory_linked_service_sftp" "compliant_missing" {
+}
+
+
+resource "other_resource" "coverage" {
+}
```

Comment:
I'm not sure what this test case is for.
However, I think we need a case where the name of the resource is not "azurerm_data_factory_linked_service_sftp" and which has `authentication_type = "Anonymous"`